Validators Testnet Docs (#190)
helium/docs: docs/mine-hnt/validators/wallet.mdx

---
id: validator-wallet
title: Configure a Wallet
hide_title: true
sidebar_label: Configure a Wallet
slug: /mine-hnt/validators/validator-wallet
---

import useBaseUrl from "@docusaurus/useBaseUrl";

# Configure a wallet

You’ll need an owner wallet to:
* Stake your Testnet Network Tokens (TNT)
* Receive validator node consensus group rewards

Note: you can have a single wallet for multiple validator nodes. However multiple wallets for a single validator node is not supported.

To download the wallet go [here](https://github.com/helium/helium-wallet-rs).

## Wallet commands

At any time use ```-h``` or ```--help``` to get more help for a command.

### Create a wallet

```./helium-wallet create basic --network testnet``` 

### View wallet balance and public key
```./helium-wallet info``` view public key and TNT balance. You’ll need this public key address to obtain TNT needed for staking. 

To obtain TNT tokens visit [Obtain Testnet Network Tokens (TNT)](/mine-hnt/validators/validator-tnt).

:::warning

>  Always make sure to inspect the transaction _before_ adding the **--committ**  option to sign and submit it to the api. There is no undo.

Reviewer: Typo: --committ

Author: fixed, thanks

* Stake the TNT:
    * ```./bin/helium-wallet validators stake <public-address> <stake> --commit```

### Unstake tokens
If you unstake tokens the validator node enters a cooldown period. During this cooldown period the validator node will not earn rewards and at the end of the cooldown period (currently 250,000 blocks or approximately 5 months) your initial staking amount will be returned to the wallet. 

#### To unstake tokens from a validator node perform the following:
```./bin/helium-wallet unstake <public-address>```

### Validator transfers 
A wallet can be paired with multiple validator nodes. Validator nodes can be transferred to different wallets. Staking amount can also be transferred to someone else.

#### To transfer a validator's stake to a new validator, or transfer the stake to a validator in another wallet:
```./bin/helium-wallet validators transfer create --new-address <new-public-address> --old-address <old-public-address>```

To transfer the stake to someone else:

```./bin/helium-wallet --format json validators transfer --new-address <new-public-address> --old-address <old-public-address> --new-owner <new-owner-public-address>```

> This will produce an output similar to the following:

Author: hey @abhay can you provide an example of actual output? Evan provided this dummy data, but as you can see the new and old addresses identical.

---
id: validator-wallet
title: Configure a Wallet
hide_title: true
sidebar_label: Configure a Wallet
slug: /mine-hnt/validators/validator-wallet
---

import useBaseUrl from "@docusaurus/useBaseUrl";

> # Configure a wallet

Reviewer 2: ```suggestion
# Configure a Wallet
```

You’ll need an owner wallet to:
* Stake your Testnet Network Tokens (TNT)
* Receive validator node consensus group rewards

> Note: you can have a single wallet for multiple validator nodes. However multiple wallets for a single validator node is not supported.

Reviewer 2: ```suggestion
:::note

You can have a single wallet for multiple validator nodes. However multiple wallets for a single validator node is not supported.

:::
```

For instructions to build the wallet go [here](https://github.com/helium/helium-wallet-rs).

At any time use ```-h``` or ```--help``` to get more help for a command.

> ## Wallet commands

Reviewer 2: ```suggestion
## Wallet Commands
```

> ### Create a wallet

Reviewer 2: ```suggestion
### Create a Wallet
```

```./bin/helium-wallet create basic --network testnet``` 

### View wallet balance and public key
```./bin/helium-wallet info``` view public key and TNT balance. You’ll need this public key address to obtain TNT needed for staking. 

To obtain TNT tokens visit [Obtain Testnet Network Tokens (TNT)](/mine-hnt/validators/validator-tnt).

:::warning

 Always make sure to inspect the transaction _before_ adding the **--committ**  option to sign and submit it to the api. There is no undo.

:::

> ### Stake tokens

Reviewer 2: ```suggestion
### Stake Tokens
```

> ### Stake tokens

Reviewer 2: I think this section should be expanded a bit to explain the staking mechanics. E.g 

* 10000 TNT to stake
* TNT is stored in wallet A, and is staked using a transaction but not actually sent to the validator node itself. 
* Etc

After you have received your TNT you need to stake these tokens. 

As part of the staking process the validator node address needs to be in the staking transaction and also needs to sign the transaction. After a wallet stakes a validator node, the wallet becomes that node’s owner, has control over that validator node, and receives rewards.

#### To stake tokens perform the following:

* Obtain the validator node address:
    * On the validator, run ```miner peer addr```.  This will produce something like: ```/p2p/112sSfUCQrVNAfFKVZWUSCeLs67wH8o1xz4yj4nKSPJKBT8bUkXi```

    In this example the part following the final slash is the validator node's public address: ```112sSfUCQrVNAfFKVZWUSCeLs67wH8o1xz4yj4nKSPJKBT8bUkXi```.

* Stake the TNT:
    * ```./bin/helium-wallet validators stake <public-address> 10000 --commit```

> ### Unstake tokens

Reviewer 2: ```suggestion
### Unstake Tokens
```

If you unstake tokens the validator node enters a cooldown period. During this cooldown period the validator node will not earn rewards and at the end of the cooldown period (currently 250,000 blocks or approximately 5 months) your initial staking amount will be returned to the wallet. The cooldown period will be shortened for the testnet to 360 blocks (approx. an hour).

#### To unstake tokens from a validator node perform the following:
```./bin/helium-wallet unstake <public-address> --commit```

### Validator transfers 
A validator node's stake can be transferred to other validator nodes in the same wallet or a different one. 

If the stake is transferred to a validator node in a different wallet, the owner of the other wallet must accept the transfer to complete the transaction.

#### To transfer stake to a new validator node in the same wallet:
```./bin/helium-wallet validators transfer create --new-address <new-public-address> --old-address <old-public-address> --commit```

#### To transfer stake to a new validator node in a different wallet:

> ```./bin/helium-wallet --format json validators transfer --new-address <new-public-address> --old-address <old-public-address> --new-owner <new-owner-wallet-public-address>```

Reviewer 3: ```suggestion
```./bin/helium-wallet --format json validators transfer create --new-address <new-public-address> --old-address <old-public-address> --new-owner <new-owner-wallet-public-address>```
```

Author: updated

This will produce an output similar to the following:

```

> {
>   "amount": 0.0,
>   "fee": 0,
>   "hash": null,
>   "new_address": "1YwbuihaJrCxxHMi7Cxu5u57W8LFpfD4J2N6NJLMCYZ2FbQ5hNg",
>   "new_owner": "1ZxcEMNnUUwujudHSfYMS9YUoQPQHor6wceRLxdpL8txWCJ7UKZ",
>   "nonce": 2,
>   "old_address": "1YwbuihaJrCxxHMi7Cxu5u57W8LFpfD4J2N6NJLMCYZ2FbQ5hNg",
>   "old_owner": "1ZxcEMNnUUwujudHSfYMS9YUoQPQHor6wceRLxdpL8txWCJ7UKZ",
>   "txn": "8gHQAQohEG6PJ7AZkUSGBqEO89CmdQAu8Y6SepBx11wgw2IY7WWGEiEQbo8nsBmRRIYGoQ7z0KZ1AC7xjpJ6kHHXXCDDYhjtZYYaIRD0idqz8SRyoUMBZhygY/fgpX9juEX+46Pya8ZyMMxa0yIhEdxxIOJxUVrnbQo2f/htY/kWQy3bWq5o0hVz4DQMZdPzMkBmtCjmV++wGbLp5oSVoUQJMYCE2u4AuhBhTZFWa2vhew6XBTYY4UIdqGiK0Fb9AD5qed7OW18smUP66Yfg+swISAI="
> }

Reviewer 3: ```suggestion
{
  "amount": 0.0,
  "fee": 60000,
  "hash": null,
  "new_address": "1YeCAsuBq3fDTv8x4iB1fJn9zBbopHhjhXGpvVwN43JDptqMT11",
  "new_owner": "1bBTWn843DwTXTyH3hbSMmDoJXHCU1D1U8cAYRdVE5rDnaS7fLb",
  "old_address": "1ZFavPP8drcqAXJPT6mLuZQqwjWNhWGr9vJ3Ajs3uJyFGW7sUMX",
  "old_owner": "1aXQ6WyNkibzhHXCFYiJQAa3BEkJnbZFsuezo4qjpW7FAgDw9Xx",
  "txn": "8gHSAQohEJdkRqMXfms5yLRyFcS+0Is8NJ4N7eLXANpGQO5j6iTHEiEQRwd/IDnFy67A2NbC3IyYvLZUrsAojyJH85aeNi3SLA0aIRE+/nQNU16ZXpbzMmszzlSw+AiGAXeKg0q7kx1DdLlY7yIhEZVpU2QzwJ1SmikaTlSEyKCQ5mCRx/xOIUAsw5MYPIsWKkD3+X75w1+267InX30CD077Y8X3JK31FEg/X1bSpFehIQa6eY/DqfDEfDjLyq6UHpxsfBEE8M0zGs9okipefnAOOODUAw=="
}
```

Author: thanks

```

You'll then need to send the txn value (a base64 string) to the new owner to be accepted. 

#### To accept the transfer the new owner performs the following:

```./bin/helium-wallet validators transfer accept --commit <txn>```

> More general wallet commands here: https://github.com/helium/helium-wallet-rs

Reviewer 2: ```suggestion
The full set of wallet commands [can be found here](https://github.com/helium/helium-wallet-rs).
```
